src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,61 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './FindMovie.scss';
+import classNames from 'classnames';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { MovieData } from '../../types/MovieData';
+import { MovieCard } from '../MovieCard';
+
+interface Props {
+  addMovie: (movie: Movie) => void,
+}
+
+export const FindMovie: React.FC<Props> = (
+  { addMovie },
+) => {
+  const [query, setQuery] = useState('');
+  const [movie, setMovie] = useState<Movie>();
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+
+  const handlSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setLoading(true);
+
+    getMovie(query)
+      .then(response => {
+        if ('Error' in response) {
+          return Promise.reject();
+        }
+
+        const {
+          Poster,
+          Title,
+          Plot,
+          imdbID,
+        } = response as MovieData;
+
+        setMovie({
+          title: Title,
+          description: Plot,
+          imgUrl: Poster === 'N/A'
+            ? 'https://via.placeholder.com/360x270.png?text=no%20preview'
+            : Poster,
+          imdbUrl: `https://www.imdb.com/title/${imdbID}`,
+          imdbId: imdbID,
+        });
+
+        return Promise.resolve();
```

Comment:
You don't need it - you didn't save the result of this promise in any variable
your `then` callback might be a `voice` type

Reply:
changed the code